uring/src/main/scala/fs2/io/uring/net/UringDatagramSocket.scala

```diff
@@ -35,7 +36,45 @@ private[net] final class UringDatagramSocket[F[_]](
   private[this] def recvmsg(msg: Ptr[msghdr], flags: Int): F[Int] =
     ring.call(io_uring_prep_recvmsg(_, fd, msg, flags))
 
-  def read: F[Datagram] = ???
+  def read: F[Datagram] =
+    readMutex.lock.surround {
+      for {
+        buf <- buffer.get(defaultReadSize)
+
+        iov <- F.delay {
+          val iov = stackalloc[uio.iovec]()
```

Comment:
So the trick with stack allocations, is that they only exist for the duration of the function in which they are created. So in this case, the `iov` allocation will actually go away as soon as the `delay` completes, because a `delay(...)` is actually a method `delay(() => ...)`.

I think we can allocate the `iovec` and `msghdr` structures at the same time as we allocate the read buffer when we create the `DatagramSocket`. Because we can only have one read in progress at a time, we will only need one and the we can re-use it.